src/game/Utils/Event_Pump.cc

```diff
@@ -10,62 +7,121 @@
 #include "Debug.h"
 #include "Logger.h"
 
-static void AddGameEventToQueue(UINT32 uiEvent, UINT16 usDelay, PTR pEventData, EventQueueID ubQueueID);
+#include <stdexcept>
+#include <vector>
 
+struct EVENT
+{
+	UINT32 TimeStamp;
+	UINT16 usDelay;
+	bool eventExpired;
+	GAMEEVENT gameEvent;
+};
 
-void AddGameEvent(GameEvent const uiEvent, UINT16 const usDelay, PTR const pEventData)
+enum class EventQueueID
 {
-	if (usDelay == DEMAND_EVENT_DELAY)
-	{
-		SLOGD("AddGameEvent: Sending Local #%d", uiEvent);
-		AddGameEventToQueue(uiEvent, 0, pEventData, DEMAND_EVENT_QUEUE);
-	}
-	else
-	{
-		SLOGD("AddGameEvent: Sending Local #%d", uiEvent);
-		AddGameEventToQueue(uiEvent, usDelay, pEventData, PRIMARY_EVENT_QUEUE);
-	}
+	PRIMARY_EVENT_QUEUE,
+	SECONDARY_EVENT_QUEUE,
+	DEMAND_EVENT_QUEUE
+};
+
+
+typedef std::vector<EVENT*>EventList;
```

Comment:
Would [std::deque](http://www.cplusplus.com/reference/deque/deque/) be a better candidate for the data structure here?

Reply:
You mean because we usually remove the first container element? Maybe, but the largest queue size I've seen was 20 and I had to burst fire an entire clip to get the number that high. OTOH [cppreference](https://en.cppreference.com/w/cpp/container/deque) has this to say about deque:

> typical implementations use a sequence of individually allocated fixed-size arrays, with additional bookkeeping, which means indexed access to deque must perform two pointer dereferences, compared to vector's indexed access which performs only one. 
>....  deques typically have large minimal memory cost; a deque holding just one element has to allocate its full internal array (e.g. 8 times the object size on 64-bit libstdc++; 16 times the object size or 4096 bytes, whichever is larger, on 64-bit libc++). 

Overall I don't believe deque is the better choice here.